parquet-hadoop/src/main/java/org/apache/parquet/filter2/compat/RowGroupFilter.java

```diff
@@ -40,15 +45,34 @@
 public class RowGroupFilter implements Visitor<List<BlockMetaData>> {
   private final List<BlockMetaData> blocks;
   private final MessageType schema;
+  private final List<FilterLevel> levels;
+  private final Supplier<FSDataInputStream> streamSupplier;
+
+  public enum FilterLevel {
+    STATISTICS,
+    DICTIONARY
+  }
 
   public static List<BlockMetaData> filterRowGroups(Filter filter, List<BlockMetaData> blocks, MessageType schema) {
     checkNotNull(filter, "filter");
     return filter.accept(new RowGroupFilter(blocks, schema));
   }
 
+  public static List<BlockMetaData> filterRowGroups(List<FilterLevel> levels, Filter filter, List<BlockMetaData> blocks, MessageType schema, Supplier<FSDataInputStream> streamSupplier) {
```

Comment:
I don't really love this signature due to exposing the data stream as a parameter.  We could reduce the entire interface to a single FilterContext, which we could then use to differentiate between the hadoop implementation and the core parquet packages.  However, the DictionaryFilter does rely on seek, which is specific to the HDFS api.